File: android/app/src/main/java/com/example/gpslogger/model/LogPoint.kt
package com.example.gpslogger.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.UUID

/** GPS/WiFiで取得した1件の位置情報ログ */
@Entity(
    tableName = "log_points",
    foreignKeys = [ForeignKey(
        entity = Session::class,
        parentColumns = ["id"],
        childColumns = ["sessionId"],
        onDelete = ForeignKey.CASCADE  // セッション削除時にログも自動削除
    )],
    indices = [Index("sessionId")]
)
data class LogPoint(
    @PrimaryKey val id: String = UUID.randomUUID().toString(),
    val sessionId: String,
    val timestamp: Long = System.currentTimeMillis(),  // Unix時刻（ミリ秒）
    val latitude: Double,
    val longitude: Double,
    val altitude: Double? = null,
    val accuracy: Float,
    val speed: Float? = null,
    val sourceType: SourceType = SourceType.GPS
)

enum class SourceType { GPS, WIFI }

File: ios/GPSLogger/Models/LogPoint.swift
import Foundation
import CoreData

enum SourceType: String, Codable {
    case gps = "GPS"
    case wifi = "WIFI"
}

/// GPS/WiFiで取得した1件の位置情報ログ
struct LogPoint: Identifiable, Codable {
    let id: UUID
    let sessionId: UUID
    let timestamp: Date
    let latitude: Double
    let longitude: Double
    let altitude: Double?
    let accuracy: Double
    let speed: Double?
    let sourceType: SourceType

    init(sessionId: UUID, latitude: Double, longitude: Double,
         altitude: Double? = nil, accuracy: Double,
         speed: Double? = nil, sourceType: SourceType = .gps) {
        self.id = UUID()
        self.sessionId = sessionId
        self.timestamp = Date()
        self.latitude = latitude
        self.longitude = longitude
        self.altitude = altitude
        self.accuracy = accuracy
        self.speed = speed
        self.sourceType = sourceType
    }
}
